### PR TITLE
Reconnect interop spec

### PR DESCRIPTION
**Pull request by yang-g**

### CODE REVIEW

**Comment by yang-g** [comment]
#2656 is the C++ client and server implementation.

**Comment by yang-g** [comment]
@a11r  PTAL

File: doc/connection-backoff-interop-test-description.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+languages.
+
+Client
+------
+
+Clients should accept these arguments:
+* --server_control_port=PORT
+    * The server port to connect to for rpc. For example, "8080"
+* --server_retry_port=PORT
+    * The server port to connect to for testing backoffs. For example, "8080"
```
Comment (a11r) [inline doc/connection-backoff-interop-test-description.md]: Suggest using different port numbers in your examples (being pedantic, but this avoids unnecessary confusion for unfamiliar readers)

File: doc/connection-backoff-interop-test-description.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+   test.
+5. Optionally, the client can do its own check on the returned backoffs.
+
+
+Server
+------
+
+A C++ server can be used for the test. Other languages do NOT need to implement
+a server. To minimize the network delay, the server binary should run on the
+same machine with the client binary.
```
Comment (a11r) [inline doc/connection-backoff-interop-test-description.md]: ... should run on the same machine or on a nearby machine (in terms of network distance) ...

File: doc/connection-backoff-interop-test-description.md (hunk 1 above)
Comment (yang-g) [inline doc/connection-backoff-interop-test-description.md]: Both done.

File: doc/connection-backoff-interop-test-description.md (hunk 2 above)
Comment (yang-g) [inline doc/connection-backoff-interop-test-description.md]: Done.

**Pull request merged**